js/ui/windowManager.js

```diff
@@ -135,6 +137,76 @@ const DisplayChangeDialog = new Lang.Class({
     },
 });
 
+const EOSShellWobbly = new Lang.Class({
+    Name: 'EOSShellWobbly',
+    Extends: EndlessShellFX.Wobbly,
+    Properties: {
+        'settings' : GObject.ParamSpec.object('settings',
+                                              'Effect GSettings',
+                                              'Wobbly Effect GSettings',
+                                              GObject.ParamFlags.CONSTRUCT_ONLY |
+                                              GObject.ParamFlags.WRITABLE,
+                                              Gio.Settings)
+    },
+
+    _init: function(params) {
+        this.parent(params);
+        this._settings = null;
+    },
+
+    set settings (val)
+    {
+         if (val['schema-id'] !== 'org.gnome.shell')
+             throw Error('EOSShellWobbly: Injected schema must be of ' +
+                         'org.gnome.shell');
+
+         this._settings = val;
+
+         let binder = Lang.bind (this, function(key, prop) {
+             this._settings.bind (key, this, prop, Gio.SettingsBindFlags.GET);
+         });
+
+         /* Bind to effect properties */
+         binder('wobbly-spring-k', 'spring-k');
+         binder('wobbly-spring-friction', 'friction');
+         binder('wobbly-slowdown-factor', 'slowdown-factor');
+         binder('wobbly-object-movement-range', 'object-movement-range');
+    },
+
+    grabbedByMouse: function() {
+
+        if (!global.get_settings().get_boolean('wobbly-effect'))
+            return;
+
+        let position = global.get_pointer();
+        let actor = this.get_actor();
+        this.grab(position[0], position[1]);
+
+        this._lastPosition = actor.get_position();
+        this._positionChangedId =
+            actor.connect('allocation-changed', Lang.bind(this, function (actor) {
```

Comment:
An interesting point actually - I've noticed that it takes movement by 3-4 pixels for the effect to "kick in" and I'm wondering if this is the cause. I'm going to give `notify::position` a try now - perhaps it might fix that one last annoying glitch.